imap_data_access/file_validation.py

```diff
@@ -68,6 +69,51 @@ def __init__(self, filename: str | Path, data_dir: Path | None = None):
         if self.error_message:
             raise self.InvalidScienceFileError(f"{self.error_message}")
 
+    @classmethod
+    def generate_from_inputs(
+        cls,
+        instrument: str,
+        level: str,
+        descriptor: str,
+        start_time: str,
+        end_time: str,
+        version: str,
+    ):
+        """Generate a filename from given inputs and return a ScienceFilePath instance.
+
+        This can be used instead of the __init__ method to make a new instance:
+        ```
+        science_file_path = ScienceFilePath.generate_from_inputs("mag", "l0", "test",
+            "20240213", "20240213", "v01-01")
+        full_path = science_file_path.construct_path()
+        ```
```

Comment:
Very cool!

For someone who doesn't like python so much, you sure are good at it 😂

Reply:
Flatterer 😆 I guess I take writing good python as a challenge